src/query/storages/fuse/src/operations/commit.rs

```diff
@@ -491,6 +509,44 @@ impl FuseTable {
     }
 }
 
+pub enum Conflict {
+    Unresolvable,
+    // resolvable conflicts with append only operation
+    // the range embedded is the range of segments that are appended in the latest snapshot
+    ResolvableAppend(Range<usize>),
+}
+
+// wraps a namespace, to clarify the who is detecting conflict
+pub struct MutatorConflictDetector;
+
+impl MutatorConflictDetector {
+    // detects conflicts, as a mutator, working on the base snapshot, with latest snapshot
+    pub fn detect_conflicts(base: &TableSnapshot, latest: &TableSnapshot) -> Conflict {
+        if base.snapshot_id == latest.snapshot_id {
```

Comment:
There is no condition that two snapshots have the same snapshot_id, maybe it can be deleted?

Reply:
right, it is too defensive